dockerfiles/services/request-replayer/src/index.php

```diff
@@ -0,0 +1,124 @@
+}
+
+define('REQUEST_LATEST_DUMP_FILE', getenv('REQUEST_LATEST_DUMP_FILE') ?: (sys_get_temp_dir() . '/dump.json'));
+define('REQUEST_LOG_FILE', getenv('REQUEST_LOG_FILE') ?: (sys_get_temp_dir() . '/requests-log.txt'));
+
+function logRequest($message, $data = '')
+{
+    if (!empty($data)) {
+        $message .= ":\n" . $data;
+    }
+    error_log(
+        sprintf('[%s | %s] %s', $_SERVER['REQUEST_URI'], REQUEST_LATEST_DUMP_FILE, $message)
+    );
+}
+
+set_error_handler(function ($number, $message) {
+    logRequest('Triggered error ' . $number . ' ' . $message);
+    trigger_error($message, $number);
+});
+
+switch ($_SERVER['REQUEST_URI']) {
+    case '/replay':
+        if (!file_exists(REQUEST_LATEST_DUMP_FILE)) {
+            logRequest('Cannot replay last request; request log does not exist');
+            break;
+        }
+        $request = file_get_contents(REQUEST_LATEST_DUMP_FILE);
+        echo $request;
+        unlink(REQUEST_LATEST_DUMP_FILE);
+        unlink(REQUEST_LOG_FILE);
+        logRequest('Returned last request and deleted request log', $request);
+        break;
+    case '/clear-dumped-data':
+        if (!file_exists(REQUEST_LATEST_DUMP_FILE)) {
+            logRequest('Cannot delete request log; request log does not exist');
+            break;
+        }
+        unlink(REQUEST_LATEST_DUMP_FILE);
+        unlink(REQUEST_LOG_FILE);
+        logRequest('Deleted request log');
+        break;
+    default:
+        $headers = getallheaders();
+        if (isset($headers['X-Datadog-Diagnostic-Check'])) {
+            logRequest('Received diagnostic check; ignoring');
+            break;
+        }
+
+        $raw = file_get_contents('php://input');
+        if (isset($headers['Content-Type']) && $headers['Content-Type'] === 'application/msgpack') {
+            // We unpack in two phases:
+            //  1) using UnpackOptions::BIGINT_AS_GMP and only asserting that trace_id, span_id and parent_id are either
+            //     integers (when <= PHP_INT_MAX) or GMP (when > PHP_INT_MAX);
+            //  2) using UnpackOptions::BIGINT_AS_STR and storing the actual result.
+            // We cannot use the first unpacked payload as when, later, we json_encode() the payload, GMPs larger than
+            // PHP_INT_MAX would be serialized to PHP_INT_MAX
+            $gmpUnpacker = new BufferUnpacker($raw, UnpackOptions::BIGINT_AS_GMP);
+            $gmpTraces = $gmpUnpacker->unpack();
+            foreach ($gmpTraces as $trace) {
+                foreach ($trace as $span) {
+                    foreach (['trace_id', 'span_id', 'parent_id'] as $field) {
+                        if (!isset($span[$field])) {
+                            continue;
+                        }
+
+                        $value = $span[$field];
+                        if (!is_int($value) && !is_a($value, 'GMP')) {
+                            logRequest("Wrong type for $field: " . var_export($value, 1));
+                            exit();
```

Comment:
Do you think it's useful to set an empty body that will serialize to the dump file below instead of exiting? If we exit when the payload is malformed (e.g. `{"span_id": "oops"}`) and then a test receives no replayed request from `request-replayer`, we might think there was an error in the transport vs an error in the payload. Or WDYT?